packaging_automation/common_tool_methods.py

```diff
@@ -0,0 +1,168 @@
+import re
+import subprocess
+from datetime import datetime
+from typing import Dict, List
+import os
+
+from github import Repository, PullRequest
+from jinja2 import Environment, FileSystemLoader
+
+from .common_validations import *
+from git import Repo
+import pathlib2
+
+BASE_GIT_PATH = pathlib2.Path(__file__).parents[1]
+
+
+def get_spec_file_name(project_name: str) -> str:
+    return f"{project_name}.spec"
+
+
+def get_version_number(version: str, fancy: bool, fancy_release_count: int) -> str:
+    fancy_suffix = f"-{fancy_release_count}" if fancy else ""
+    return f"{version}{fancy_suffix}"
+
+
+def get_project_version_from_tag_name(tag_name: is_tag(str)) -> str:
+    return tag_name[1:]
+
+
+def get_version_number_with_project_name(project_name: str, version: str, fancy: bool, fancy_release_count: int) -> str:
+    fancy_suffix = f"-{fancy_release_count}" if fancy else ""
+    return f"{version}.{project_name}{fancy_suffix}"
+
+
+def get_template_environment(template_dir: str) -> Environment:
+    file_loader = FileSystemLoader(template_dir)
+    env = Environment(loader=file_loader)
+    return env
+
+
+def find_nth_occurrence_position(subject_string: str, search_string: str, n) -> int:
+    start = subject_string.find(search_string)
+
+    while start >= 0 and n > 1:
+        start = subject_string.find(search_string, start + 1)
+        n -= 1
+    return start
+
+
+def find_nth_matching_line_number(subject_string: str, regex_pattern: str, n: int) -> int:
+    """Takes a subject string, regex param and the search index as parameter and returns line number of found match.
+    If not found returns -1"""
+    lines = subject_string.splitlines()
+    counter = 0
+    for line_number, line in enumerate(lines):
+        if re.match(regex_pattern, line):
+            counter = counter + 1
+        if counter == n:
+            return line_number
+    return -1
+
+
+def find_nth_matching_line(subject_string: str, regex_pattern: str, n: int) -> str:
+    """Takes a subject string, regex param and the search index as parameter and returns line content of found match.
+        If not found returns empty string"""
+    lines = subject_string.splitlines()
+
+    line_number = find_nth_matching_line_number(subject_string, regex_pattern, n)
+    if line_number != -1:
+        return lines[line_number]
+    else:
+        return ""
+
+
+def remove_string_inside_parentheses(param: str) -> str:
+    return re.sub(r"[(\[].*?[)\]]", "", param)
```

Comment:
1) seems that this removes the parenthesis too (since the second argument is `""`)

2) parentheses -> parenthesis

3) putting an example for that via comment would be useful: # e.g: [string] -> []

Reply:
Changed method name into remove_text_with_parenthesis
Put docstring onto method

Reply:
two minor comments:

* Removes texts within parenthesis together with the parenthesis itself ..
* parentheses -> parenthesis

Reply:
Fixed